nixos/modules/services/x11/xserver.nix

```diff
@@ -651,25 +651,33 @@ in
     system.extraDependencies = singleton (pkgs.runCommand "xkb-layouts-exist" {
       inherit (cfg) layout xkbDir;
     } ''
-      if sed -n -e ':i /^! \(layout\|variant\) *$/ {
-          :l; n; /^!/bi; s/^ *\([^ ]\+\).*/\1/p; tl
-         }' "$xkbDir/rules/base.lst" | grep -qxF "$layout"
-      then
-        touch "$out"
+      IFS=, read -a layouts <<<'${cfg.layout}'
+
+      missing=()
+      for layout in "''${layouts[@]}" ; do
+        if ! sed -n -e ':i /^! \(layout\|variant\) *$/ {
+            :l; n; /^!/bi; s/^ *\([^ ]\+\).*/\1/p; tl
+           }' "$xkbDir/rules/base.lst" | grep -qxF "$layout"
+        then
+          missing+=("$layout")
+        fi
+      done
+
+      if [[ ''${#missing[@]} = 0 ]] ; then
+        touch $out
         exit 0
       fi
-
       cat >&2 <<-EOF
 
-      The selected keyboard layout definition does not exist:
+      The selected keyboard layout definitions do not exist:
```

Comment:
Good that you are trying to help the user, but you could also just output all the valid values (a constructive reply from the machine) with the file name, instead of letting the machine say that the user messed up without any help.